File: fixed_point_conversions/generate_widthcr.c
#include "settings.h"
#include <string>
#include <iostream>
#include <bitset>
#include <cmath>

#define DEBUG 0

// NOTE: This code isn't generating WidthCr but W_Cr/WidthCr. Bad filename, I know...

int main()
{
	for (int i = 0; i < 256; i++)
	{
		double Y = i;
		double output = W_Cr / WidthCr(Y);
		
		if (DEBUG)	
			std::cout << "Should be: " << output << " is: " << convert_from_fixed_point(convert_to_fixed_point(output)) << std::endl;
		std::cout << convert_to_fixed_point(output).to_string() << std::endl;
	}
}

File: fixed_point_conversions/generate_widthcr.py
from settings import *

for i in range(2**8):
	Y = i
	output = 0

	if Y <= K_l:
		output = WL_Cr + (Y - Y_min)*(W_Cr - WL_Cr)/(K_l - Y_min)
	elif K_h <= Y:
		output = WH_Cr + (Y_max - Y)*(W_Cr - WH_Cr)/(Y_max - K_h)
	else:
		output = 0.0;

	if output != 0: 
		output = W_Cr / output		
		
	output = convert_to_fixed_point(output)

	if output == 0:
		print("0"*width)
	else:
		print("{:b}".format(output).zfill(width))